### ISSUE TITLE
Make release process smoother

### ISSUE THREAD

**Issue by Riduidel** [Riduidel/rrss2imap#15]
Currently, I only use `cargo release` to push releases to crates.io.
This is far from enough.
I want to have a release process as smooth as what I'm used to in other languages (typically Java).

So I would like a release process which

1. push sources to crates.io
2. tag the git tree
3. get the changelog (using [`git journal`](https://github.com/saschagrunert/git-journal))
3. push the release to GitHub release with the provided changelog (I guess it's a job for [`cargo hublish`](https://github.com/chasinglogic/cargo-hublish))

I think it's complicated enough to require an orchestration script. I could use the classical [`build.rs`](https://doc.rust-lang.org/cargo/reference/build-scripts.html) ... But I find more fun to use [`cargo make`](https://github.com/sagiegurari/cargo-make#usage-predefined-flows)

So, to recap, I will create a cargo make script that will launch cargo release, git journal and cargo hublish (in that order) with some infos communicated in between).

**Comment by Riduidel** [comment]
How crazy, I've never reealized cargo release already pushes zips/.tar.gz to GitHub ! I now have two questions

1. How to have git journal used as commit message ?
2. How to have binary artifacts pushed ?

**Comment by Riduidel** [comment]
For commit message, what if I use [pre-release-replacements](https://github.com/sunng87/cargo-release#maintaining-changelog) to have git journal changelog injected ?

**Comment by Riduidel** [comment]
See [issue 86](https://github.com/sunng87/cargo-release/issues/86) for details.

**Comment by Riduidel** [comment]
All this is useless, now I use travis-ci, which is indeed way smoother
